fw/http.c

```diff
@@ -5758,6 +5733,28 @@ tfw_h1_req_process(TfwStream *stream, struct sk_buff *skb)
 	return hmsib;
 }
 
+static void
+extract_req_host(TfwHttpReq *req)
+{
+	int hid = 0;
+	TfwStr *hdrs = req->h_tbl->tbl;
+
+	if (TFW_MSG_H2(req)) {
+		if (!TFW_STR_EMPTY(&hdrs[TFW_HTTP_HDR_H2_AUTHORITY]))
+			hid = TFW_HTTP_HDR_H2_AUTHORITY;
+		else
+			hid = TFW_HTTP_HDR_HOST;
+		__h2_msg_hdr_val(&hdrs[hid], &req->host);
+	} else {
+		if (TFW_STR_EMPTY(&req->host)) {
+			hid = TFW_HTTP_HDR_HOST;
+			tfw_http_msg_clnthdr_val(req, &hdrs[TFW_HTTP_HDR_HOST],
+						 TFW_HTTP_HDR_HOST,
+						 &req->host);
+		}
+	}
```

Comment:
So the authority part of URI takes priority over `Host` header in HTTP/1? Could you please cite RFC which describes this rule as well as for HTTP/2 precedence of `:authority` over the `Host` header.

We discussed that we should get rid of `http_host_required` frang limit and make the rule default. This could mean that we just should call `frang_http_host_check()` instead of this function and set `req->host` there.

We actually do not work without the limit: e.g. `tfw_h2_adjust_req()` just returns error on `(!auth && !host)` - this code can be eliminated. Probably there are other similar places.

Once we have mandatory Host and URI authority equal the task https://github.com/tempesta-tech/tempesta/issues/1667 should be fixed, so I assign it to you as well.

Reply:
>We discussed that we should get rid of http_host_required frang limit and make the rule default

As I can see from the code, this rule also enforces equality of passed authorities which significantly changes tempesta-fw behavior. In particular we have tests that send different values in Host and forwarded headers and expect match on the Forwarded host value. Enforcing frang rule would break such tests.
At the same time new code doesn't introduce any pitfalls on mismatching absolute-uri/host/authority/forwarded headers, so maybe we should make a separate issue on that?
Also note that updated RFC explicitly mentioning that proxy which accepts absolute-uri MUST replace the Host headers if they don't match in the incoming request: ttps://www.rfc-editor.org/rfc/rfc7230#section-5.4
>When a proxy receives a request with an absolute-form of request-target, the proxy MUST ignore the received Host header field (if any) and instead replace it with the host information of the request-target.  A proxy that forwards such a request MUST generate a new Host field-value based on the received request-target rather than forward the received Host field-value.

Reply:
This comment is still not addressed

Reply:
This particular function transforms H2 into H1 and H2 cannot have absolute URI, so relying on authority/host is completely fine. And as these headers may change after `tfw_h2_req_set_loc_hdrs()` invocation, this isn't a bad idea to leave this code as is.

Reply:
I believe `tfw_h2_req_set_loc_hdrs()` doesn't impact the comment because it is about the code:
```C
bool auth = !TFW_STR_EMPTY(&ht->tbl[TFW_HTTP_HDR_H2_AUTHORITY]);
bool host = !TFW_STR_EMPTY(&ht->tbl[TFW_HTTP_HDR_HOST]);
....
if (!auth && !host) {
                T_WARN("Cant convert h2 request to http/1.1: no authority "                    
                       "found\n");                                                             
                return -EINVAL;                                                                
        }     
```
which is before `tfw_h2_req_set_loc_hdrs()`. Once we checked the headers in `extract_req_host()` there is no sense to check them again